### PR TITLE
Changed World-Pixel transformation methods to be optional

### PR DESCRIPTION
**Pull request by IodaMikeMapbox**
Checks:

* [x] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
resolves mapbox/mapbox-vision#1049

### CODE REVIEW

**Comment by dersim-davaod** [comment]
@IodaMikeMapbox, agree with @chezzdev above ^^.
Additionally you should use `do not merge` label in such cases.
Moreover could you please assign appropriate labels for PRs (at least `iOS` to see PR in ZenHub?

File: MapboxVision/VisionManager/VisionManagerProtocol.swift [hunk 1]
```diff
@@ -16,14 +16,14 @@ public protocol VisionManagerProtocol: AnyObject {
 
      - Parameter screenCoordinate: Screen coordinate expressed in pixels
      */
-    func pixelToWorld(screenCoordinate: Point2D) -> WorldCoordinate
+    func pixelToWorld(screenCoordinate: Point2D) -> WorldCoordinate?
 
     /**
      Converts location of the point from a world coordinate to a screen coordinate.
 
      - Parameter worldCoordinate: Point in world coordinate
```
Comment (dersim-davaod) [inline MapboxVision/VisionManager/VisionManagerProtocol.swift]: Should we describe when update methods here and above returns `nil`?

Comment (IodaMikeMapbox) [reply MapboxVision/VisionManager/VisionManagerProtocol.swift]: fine

File: MapboxVision/VisionManager/VisionManagerProtocol.swift [hunk 2]
```diff
@@ -15,15 +15,17 @@ public protocol VisionManagerProtocol: AnyObject {
...
+     - Returns: world coordinate if it could be projected on road and nil otherwise
      */
-    func pixelToWorld(screenCoordinate: Point2D) -> WorldCoordinate
+    func pixelToWorld(screenCoordinate: Point2D) -> WorldCoordinate?
 
     /**
      Converts location of the point from a world coordinate to a screen coordinate.
 
      - Parameter worldCoordinate: Point in world coordinate
+     - Returns: screen coordinate if `worldCoordinate` could be represented in screen coordinates
```
Comment (chezzdev) [inline MapboxVision/VisionManager/VisionManagerProtocol.swift]:
```suggestion
     - Returns: Screen coordinate if `worldCoordinate` can be represented in screen coordinates
```

File: MapboxVision/VisionManager/VisionManagerProtocol.swift [hunk 3]
```diff
@@ -14,15 +14,17 @@ public protocol VisionManagerProtocol: AnyObject {
...
+     - Returns: World coordinate if `screenCoordinate` can be projected on the road and nil otherwise
      */
-    func pixelToWorld(screenCoordinate: Point2D) -> WorldCoordinate
+    func pixelToWorld(screenCoordinate: Point2D) -> WorldCoordinate?
 
     /**
      Converts location of the point from a world coordinate to a screen coordinate.
 
      - Parameter worldCoordinate: Point in world coordinate
+     - Returns: Screen coordinate if `worldCoordinate` can be represented in screen coordinates
```
Comment (chezzdev) [inline MapboxVision/VisionManager/VisionManagerProtocol.swift]: Is "... and nil otherwise" missing here?

Comment (IodaMikeMapbox) [reply MapboxVision/VisionManager/VisionManagerProtocol.swift]: fixed

File: MapboxVision/VisionManager/VisionManagerProtocol.swift (hunk 2 above)
Comment (IodaMikeMapbox) [reply MapboxVision/VisionManager/VisionManagerProtocol.swift]: done

File: MapboxVision/VisionManager/VisionManagerProtocol.swift [hunk 4]
```diff
@@ -15,15 +15,17 @@ public protocol VisionManagerProtocol: AnyObject {
      Converts location of the point from a screen coordinate to a world coordinate.
 
      - Parameter screenCoordinate: Screen coordinate expressed in pixels
+     - Returns: world coordinate if it could be projected on road and nil otherwise
```
[the remark this answers is not in the document]
Comment (IodaMikeMapbox) [reply MapboxVision/VisionManager/VisionManagerProtocol.swift]: done

**Pull request merged**